massa-execution-worker/src/speculative_roll_state.rs

```diff
@@ -476,4 +507,56 @@ impl SpeculativeRollState {
 
         credits
     }
+
+    /// Remove a given amount from deferred credits and return the removed amount
```

Comment:
Explain that if it cannot slash everytging, it will still try to slash as much as it can

Reply:
done